sdks/java/io/clickhouse/src/main/java/org/apache/beam/sdk/io/clickhouse/ClickHouseWriter.java

```diff
@@ -39,19 +41,19 @@ public class ClickHouseWriter {
   private static final Instant EPOCH_INSTANT = new Instant(0L);
 
   @SuppressWarnings("unchecked")
-  static void writeNullableValue(
-      ClickHouseRowBinaryStream stream, ColumnType columnType, Object value) throws IOException {
+  static void writeNullableValue(ClickHouseOutputStream stream, ColumnType columnType, Object value)
+      throws IOException {
 
     if (value == null) {
-      stream.markNextNullable(true);
+      BinaryStreamUtils.writeNull(stream);
     } else {
-      stream.markNextNullable(false);
+      BinaryStreamUtils.writeNonNull(stream);
```

Comment:
This api looks strange to me. Do you really need to manually write a null bit to the stream?

Reply:
In ClickHouse Nullable mean that the field is optional in that case when  we do not have data we write null and when there is data we write not null and then the value